rfc/rfc-78/rfc-78.md

```diff
@@ -0,0 +1,339 @@
+
+### FileSystemView:
+- Support ignoring partially failed log files from FSV. In 0.16.0, from FSV standpoint, all log files(including partially failed) are valid. We let the log record reader ignore the partially failed log files. But
+  in 1.x, log files could be rolledback (deleted) by a concurrent rollback. So, the FSV should ensure it ignores the uncommitted log files. 
+- We don't need the completion time logic ported over either for file slice determination nor for log file ordering.\
+    - So, here is how File Slice determination will happen using 0.16.0 reader.\
+      a. Read base files and assign to resp file groups. \
+      b. Read log files. Parse instant time from log file name (it could refer to base instant time for a file written in 0.16.x, or it could refer to delta commit in case of 1.x writer). Find largest (<=) base instant times in the corresponding file group
+      and assign the log file to it. \
+      d. Log files within a file slice are ordered based on log version and write tokens.\
+      The same logic will be used whether we are reading a 0.16.x table or its a 1.x table. Only difference wrt how a 1.x reader will behave in comparison to 0.16.x reader while reading a 1.x table is when NBCC is involved w/ multi-writers. But as of this writing, 
+  we are not supporting that in 0.16.x. 
+    - Lets see with an illustrative example.\
+      i. Table written in 0.16.x. Say total files are bf1_t10, lf1_t10, lf2_t10, bf2_t100, lf3_100, lf4_100.\
+      If we run through above algo, lf1 and lf2 will be assigned to file slice with base instant time t10,\
+  and lf3 and lf4 will be assigned to file slice with base instant time t100.\
+      ii. Table written in 1.x.   Say total files are bf1_t10, lf1_t15_{t50}, lf2_t60_{t90}, bf2_t100, lf3_t110_{t140}, lf4_t150_{t200}. Time within braces are completion times which is not really part of log file name, but showing it just for illustration.\
+      If we run through above algo, lf1 and lf2 will be assigned to file slice with base instant time t10 (since the max base instant time which is less than instant times of the log files (start time/delta commit time) is t10\
+      and lf3 and lf4 will be assigned to file slice with base instant time t100 using similar logic.\
+- FSV building/reading should account for commit instants in both 0.x and 1.x formats. Since completion time log file ordering is not supported in 0.16.0 (that’s our current assumption), we may not need to support exactly how a FSV in 1.x reader supports. But below items should be supported.
```

Comment:
1 minor correction to above logic. we need to fix log file ordering for a given file slice. for log files created in 1.x, since all of them have diff delta commits, the log file version is pretty much 1. So, our log file ordering should be based on "instant time" and then "log file version" and then "write token" in that order. 
This should take care of correct ordering for both 0.x log files and 1.x log files within a file slice.

Reply:
yes yes. we are not making any assumption. "instant time", "log file version", "write token" -> this is how our log file comparator will be designed